src/components/Offices/TaxPayers/index.js

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Card, Typography, Row, Col, Space, Select,
+} from 'antd';
+import { identityActions, officesActions } from '../../../redux/actions';
+import { officesSelectors } from '../../../redux/selectors';
+import { formatMerits } from '../../../utils/walletHelpers';
+import HistoryCopyIconWithAddress from '../../Wallet/HistoryCopyIconWithAddress';
+
+const { Title, Text } = Typography;
+const { Option } = Select;
+
+const getRankSuffix = (rank) => {
+  if (rank === 1) return 'st';
+  if (rank === 2) return 'nd';
+  if (rank === 3) return 'rd';
+  return 'th';
+};
+
+export default function TaxPayers() {
+  const [timePeriodInMonth, setTimePeriodInMonth] = useState(12);
+  const dispatch = useDispatch();
+  const taxesPayers = useSelector(officesSelectors.selectorTaxesPayers);
+  const { sortedPoolTotals, sortedUnpoolTotals, sortedTotalsByAddressPoolTotal } = taxesPayers;
+
+  const topPoolTotals = sortedPoolTotals?.slice(0, 10);
+  const topUnpoolTotals = sortedUnpoolTotals?.slice(0, 10);
+
+  useEffect(() => {
+    dispatch(officesActions.getTaxPayers.call(timePeriodInMonth));
+  }, [dispatch, timePeriodInMonth]);
+
+  useEffect(() => {
+    const poolList = sortedPoolTotals
+      ? sortedPoolTotals.map((item) => item.addressId)
+      : [];
+    const unPoolList = sortedUnpoolTotals
+      ? sortedUnpoolTotals.map((item) => item.addressId)
+      : [];
+    const topTaxpayersList = sortedTotalsByAddressPoolTotal
+      ? sortedTotalsByAddressPoolTotal.map((item) => item.addressId)
+      : [];
+    dispatch(
+      identityActions.getIdentityMotions.call(
+        Array.from(new Set(poolList.concat(unPoolList.flat()).concat(topTaxpayersList.flat()))),
+      ),
+    );
+  }, [dispatch, sortedPoolTotals, sortedUnpoolTotals, sortedTotalsByAddressPoolTotal]);
+
+  const handleSelectChange = (value) => {
+    setTimePeriodInMonth(value);
+  };
+
+  return (
+    <div>
```

Comment:
Maybe show a loader here for the data (<Spin />) before it loads + use unobtrusive loader